File: bug67_duplicate_processing_in_loop/bug67_duplicate_processing_in_loop/main1.js
function main() {
    let data = [];
    for (let i = 0n; i < 10n; i++) data.push(0x100n * (i + 1n));
    
    let acc = 0n;

    for (let j = 0; j < 10; j++) {
        let i = BigInt(j);
        let v = data[j] * 0x9E3779B97F4A7C15n;

        acc ^= v; // ✅ FIX: removed duplicate XOR
        acc &= 0xFFFFFFFFFFFFFFFFn;

        let shift = (i % 9n + 1n);
        acc = ((acc << shift) | (acc >> (64n - shift))) & 0xFFFFFFFFFFFFFFFFn; // ✅ FIX
    }

    console.log(acc.toString(16)); // ✅ Added output
}
main();

File: bug67_duplicate_processing_in_loop/bug67_duplicate_processing_in_loop/main1.java
public class main1 {
    public static void main(String[] args) {
        int[] data = new int[10];
        for (int i = 0; i < 10; i++) data[i] = 0x100 * (i + 1);
        
        long acc = 0;

        for (int i = 0; i < 10; i++) {
            long v = (long)data[i] * 0x9E3779B97F4A7C15L;

            acc ^= v; // ✅ FIX: removed duplicate XOR

            int shift = (i % 9 + 1);
            acc = (acc << shift) | (acc >>> (64 - shift)); // ✅ FIX
        }

        System.out.println(Long.toHexString(acc)); // ✅ Added output
    }
}